### PR TITLE
Fixes M2-1591, M2-1588

### PR DESCRIPTION
**Pull request by BamMironov**
https://mindlogger.atlassian.net/browse/M2-1591
https://mindlogger.atlassian.net/browse/M2-1588

### CODE REVIEW

File: src/widgets/survey/model/hooks/useFlowRecordInitialization.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      const activityIds = applet.activityFlows.find(
+        flow => flow.id === flowId,
+      )?.activityIds;
+
+      return buildActivityFlowPipeline({
+        activityIds: activityIds!,
+        appletId,
+        eventId,
+        flowId,
+        startFrom: step,
```
Comment (anq83) [inline src/widgets/survey/model/hooks/useFlowRecordInitialization.ts:58]: Will step always equial to 0? If so, then set it explicitly and remove dependency

File: src/widgets/survey/ui/Finish.tsx [hunk 2]
```diff
@@ -41,21 +41,30 @@ function FinishItem({
       appletId,
       activityId,
       eventId,
+      order,
     });
 
+  const { isTimerElapsed } = useFlowState({
```
Comment (anq83) [inline src/widgets/survey/ui/Finish.tsx:47]: This flag will not work correctly due to double usage of useState inside it

File: src/widgets/survey/ui/Finish.tsx [hunk 3]
```diff
@@ -41,21 +41,30 @@ function FinishItem({
...
+
   const {
     mutate: sendAnswers,
     isLoading: isSendingAnswers,
     isSuccess: successfullySentAnswers,
     isPaused: isOffline,
   } = useActivityAnswersMutation({
     onError: error => {
-      if (error.evaluatedMessage) {
+      if (error.response.status !== 401 && error.evaluatedMessage) {
```
Comment (anq83) [inline src/widgets/survey/ui/Finish.tsx:60]: Will it be redirected to login screen exactly at this moment ?

File: src/widgets/survey/ui/Intermediate.tsx [hunk 4]
```diff
@@ -60,13 +63,23 @@ function Intermediate({
     select: r => mapActivitiesFromDto(r.data.result.activities),
   });
 
-  const totalActivities = activityFlow!.activityIds.length;
```
Comment (anq83) [inline src/widgets/survey/ui/Intermediate.tsx:63]: useAppletDetailsQuery above should be excluded and all dependent code should relay only on pipleline
add a comment - todo at least

**Review by anq83 — CHANGES_REQUESTED** [review]

File: src/widgets/survey/model/hooks/useFlowRecordInitialization.ts (hunk 1 above)
Comment (BamMironov) [reply src/widgets/survey/model/hooks/useFlowRecordInitialization.ts:58]: No, it won't always equal 0

File: src/widgets/survey/ui/Finish.tsx (hunk 3 above)
Comment (BamMironov) [reply src/widgets/survey/ui/Finish.tsx:60]: Automatic redirection to the Logic screen is handled on `RootNavigator.tsx`
image

File: src/widgets/survey/ui/Intermediate.tsx (hunk 4 above)
Comment (BamMironov) [reply src/widgets/survey/ui/Intermediate.tsx:63]: Added a comment

**Review by anq83 — APPROVED** [review]

**Pull request merged**